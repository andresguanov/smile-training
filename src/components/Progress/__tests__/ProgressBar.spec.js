import { shallowMount, createLocalVue } from "@vue/test-utils"
import smProgressBar from "@components/Progress/ProgressBar"
let wrapper
const localVue = createLocalVue()
import VueTippy from "vue-tippy"
localVue.use(VueTippy, {
  directive: "tooltip"
})

describe("smProgressBar", () => {
  //#region Basic render
  it("renders default smProgressBar", () => {
    wrapper = shallowMount(smProgressBar, {
      localVue,
      propsData: {
        value: 20
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
  //#endregion

  //#region Props
  it("shows percentage", () => {
    wrapper = shallowMount(smProgressBar, {
      localVue,
      propsData: {
        value: 20,
        max: 200,
        showPercentage: true
      }
    })
    expect(wrapper.attributes("value")).toBe("10")
  })
  //#endregion

  //#region Methods
  //#endregion

  //#region Events
  //#endregion

  //#region Slots
  //#endregion
})
