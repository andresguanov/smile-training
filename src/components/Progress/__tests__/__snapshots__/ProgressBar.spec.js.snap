// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smProgressBar renders default smProgressBar 1`] = `<progress max="100" value="20"></progress>`;
